### ISSUE TITLE
../src/fi_tostr.c:229:2: note: in expansion of macro 'CASEENUMSTR'

### ISSUE THREAD

**Issue by jeffhammond** [ofi-cray/libfabric-cray#526]
I followed the directions on the Wiki but the build fails.

The fix is trivial - comment out line 229 in `src/fi_tostr.c` - so I won't bother to submit a patch or pull request.

```
jhammond@edison12:~/OFI/libfabric/build-gcc> module list
Currently Loaded Modulefiles:
  1) modules/3.2.10.3                      10) moab/8.0.1-2014110616-5c7a394-sles11  19) dmapp/7.0.1-1.0502.10246.8.47.ari
  2) nsg/1.2.0                             11) altd/2.0                              20) gni-headers/4.0-1.0502.10317.9.2.ari
  3) eswrap/1.1.0-1.020200.1130.0          12) darshan/2.3.0                         21) xpmem/0.1-2.0502.57015.1.15.ari
  4) switch/1.0-1.0502.57058.1.58.ari      13) craype/2.3.1                          22) dvs/2.5_0.9.0-1.0502.1958.2.55.ari
  5) craype-network-aries                  14) gcc/4.9.2                             23) alps/5.2.3-2.0502.9295.14.14.ari
  6) craype-ivybridge                      15) cray-libsci/13.0.3                    24) rca/1.0.0-2.0502.57212.2.56.ari
  7) cray-shmem/7.2.1                      16) udreg/2.3.2-1.0502.9889.2.20.ari      25) atp/1.8.2
  8) cray-mpich/7.2.1                      17) ugni/6.0-1.0502.10245.9.9.ari         26) PrgEnv-gnu/5.2.56
  9) torque/5.0.1                          18) pmi/5.0.6-1.0000.10439.140.2.ari
jhammond@edison12:~/OFI/libfabric/build-gcc> ../configure CC=gcc --disable-sockets --enable-gni && make -j12
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking for gcc option to accept ISO C99... -std=gnu99
checking whether gcc -std=gnu99 and cc understand -c and -o together... yes
checking how to run the C preprocessor... gcc -std=gnu99 -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for typeof syntax and keyword spelling... typeof
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc -std=gnu99... /usr/common/usg/altd/2.0/bin/ld
checking if the linker (/usr/common/usg/altd/2.0/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 3458764513820540925
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/common/usg/altd/2.0/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc -std=gnu99 object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc -std=gnu99 supports -fno-rtti -fno-exceptions... no
checking for gcc -std=gnu99 option to produce PIC... -fPIC -DPIC
checking if gcc -std=gnu99 PIC flag -fPIC -DPIC works... yes
checking if gcc -std=gnu99 static flag -static works... yes
checking if gcc -std=gnu99 supports -c -o file.o... yes
checking if gcc -std=gnu99 supports -c -o file.o... (cached) yes
checking whether the gcc -std=gnu99 linker (/usr/common/usg/altd/2.0/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for dlopen in -ldl... yes
checking for pthread_mutex_init in -lpthread... yes
checking for pthread_spin_init... yes
checking for library containing clock_gettime... -lrt
checking compiler support for c11 atomics... yes
checking whether ld accepts --version-script... yes
checking for .symver assembler support... yes
checking for __alias__ attribute support... yes
configure: *** Configuring psm provider
checking psm.h usability... no
checking psm.h presence... no
checking for psm.h... no
configure: psm provider: disabled
configure: *** Configuring psm2 provider
checking psm2.h usability... no
checking psm2.h presence... no
checking for psm2.h... no
configure: psm2 provider: disabled
configure: *** Configuring sockets provider
configure: sockets provider: disabled
configure: *** Configuring verbs provider
checking infiniband/verbs.h usability... yes
checking infiniband/verbs.h presence... yes
checking for infiniband/verbs.h... yes
looking for library without search path
checking for ibv_open_device in -libverbs... yes
checking rdma/rsocket.h usability... no
checking rdma/rsocket.h presence... no
checking for rdma/rsocket.h... no
configure: verbs provider: disabled
configure: *** Configuring usnic provider
checking for infiniband/verbs.h... (cached) yes
configure: checking for libnl3
checking for libnl3 prefix... /usr
checking for /usr/include/libnl3... not found
configure: checking for libnl
looking for header without includes
checking netlink/netlink.h usability... no
checking netlink/netlink.h presence... no
checking for netlink/netlink.h... no
checking netlink/netlink.h usability... no
checking netlink/netlink.h presence... no
checking for netlink/netlink.h... no
configure: usnic provider: disabled
configure: *** Configuring mxm provider
checking size of void *... 8
configure: MXM not enabled
configure: mxm provider: disabled
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
configure: *** Configuring gni provider
checking for CRAY_UGNI... yes
checking for CRAY_GNI_HEADERS... yes
checking for CRAY_ALPS_LLI... yes
checking for CRAY_ALPS_UTIL... yes
configure: gni provider: include in libfabric
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libfabric.spec
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
***
*** Built-in providers: gni 
*** DSO providers:  
***
make  all-am
make[1]: Entering directory `/global/u1/j/jhammond/OFI/libfabric/build-gcc'
  CC       src/src_libfabric_la-fabric.lo
  CC       src/src_libfabric_la-fi_tostr.lo
  CC       src/src_libfabric_la-log.lo
  CC       src/src_libfabric_la-var.lo
  CC       src/src_libfabric_la-common.lo
  CC       src/src_libfabric_la-enosys.lo
  CC       src/src_libfabric_la-rbtree.lo
  CC       src/src_libfabric_la-fasthash.lo
  CC       prov/gni/src/src_libfabric_la-gnix_atomic.lo
  CC       prov/gni/src/src_libfabric_la-gnix_av.lo
  CC       prov/gni/src/src_libfabric_la-gnix_cm.lo
  CC       prov/gni/src/src_libfabric_la-gnix_bitmap.lo
../src/fi_tostr.c: In function 'fi_tostr_protocol':
../src/fi_tostr.c:73:2: error: duplicate case value
  case SYM: { strcatf(buf, #SYM); break; }
  ^
../src/fi_tostr.c:229:2: note: in expansion of macro 'CASEENUMSTR'
  CASEENUMSTR(FI_PROTO_GNI);
  ^
../src/fi_tostr.c:73:2: error: previously used here
  case SYM: { strcatf(buf, #SYM); break; }
  ^
../src/fi_tostr.c:222:2: note: in expansion of macro 'CASEENUMSTR'
  CASEENUMSTR(FI_PROTO_GNI);
  ^
make[1]: *** [src/src_libfabric_la-fi_tostr.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/global/u1/j/jhammond/OFI/libfabric/build-gcc'
make: *** [all] Error 2
```

**Comment by sungeunchoi** [comment]
Something went funny during the mega-merge of this weekend, and @hppritcha thinks that a commit got left out.  Sorry about that, but yes easy to fix.  Waiting for Howard to take a look, so he can do some post mortem to make sure nothing else got screwed up.

**Comment by sungeunchoi** [comment]
Fixed by #528
